src/data-scraper/index.ts

```diff
@@ -7,23 +7,59 @@ import { IRuntimeImagesResponse } from '../transmitter/types';
 import { NeedleOptions } from 'needle';
 import { Agent as HttpsAgent } from 'https';
 
+const httpsAgent = new HttpsAgent({
+  keepAlive: true,
+  // We agreed with Sysdig to skip TLS certificates validation for HTTPS connection.
+  rejectUnauthorized: false,
+});
+
+function getSysdigUrl(): string {
+  const base: string = config.SYSDIG_ENDPOINT;
+  const url: string = base + '/v1/runtimeimages';
+  return url;
+}
+
+function getSysdigAuthHeader(): string {
+  return `Bearer ${config.SYSDIG_TOKEN}`;
+}
+
 function isSuccessStatusCode(statusCode: number | undefined): boolean {
   return statusCode !== undefined && statusCode >= 200 && statusCode < 300;
 }
 
+/** NOTE: This function can throw, so the caller should handle errors. */
+export async function validateConnectivity(): Promise<void> {
+  const url = getSysdigUrl();
+  const header = getSysdigAuthHeader();
+  const reqOptions: NeedleOptions = {
+    agent: httpsAgent,
+    headers: {
+      authorization: header,
+    },
+    timeout: 10_000,
+  };
+
+  const limit: number = 1;
+  const cursor: string = '';
+  const { response } = await retryRequest(
+    'get',
```

Comment:
If Sysdig supports HEAD requests, we can try that, but this is nit 🤗

Reply:
After discussing it, the current approach seems to be the recommended one